src/Microsoft.Data/DataFrame.IO.cs

```diff
@@ -0,0 +1,261 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Data
+{
+
+    public partial class DataFrame
+    {
+        private static Type GuessKind(int col, List<string[]> read)
+        {
+            Type res = typeof(string);
+            int nbline = 0;
+            foreach (var line in read)
+            {
+                if (col >= line.Length)
+                    throw new FormatException(string.Format(Strings.LessColumnsThatExpected, nbline + 1)); // Will this really work?
+
+                var val = line[col];
+                try
+                {
+                    bool.Parse(val);
```

Comment:
`TryParse` instead of `Parse` and an exception. Throwing exceptions in happy paths will cause perf degradation.

Reply:
Are you planning on profiling the CSV parsing and seeing where any bottlenecks are?

---
In reply to: [299702122](https://github.com/dotnet/corefxlab/pull/2692#discussion_r299702122) [](ancestors = 299702122)

Reply:
Yup. Sometime soon. There's a whole bunch of perf tests I want to do.